### PR TITLE
Sort media by default semantics.

### PR DESCRIPTION
**Pull request by ctolkien**
- Includes test, and VS test runner for xunit

### CODE REVIEW

File: Code/OpenRealEstate.Core/Models/Media.cs [hunk 1]
```diff
@@ -74,5 +74,31 @@ public void Copy(Media newMedia, CopyDataOptions copyDataOptions = CopyDataOptio
...
         }
+
+        public int CompareTo(Media other)
+        {
+            if (other == null)
+            {
+                return 1;
+            }
+            return Order.CompareTo(other.Order);
+
```
Comment (PureKrome) [inline Code/OpenRealEstate.Core/Models/Media.cs:85]: Delete empty line.

File: Code/OpenRealEstate.Core/Models/Media.cs [hunk 2]
```diff
@@ -74,5 +74,31 @@ public void Copy(Media newMedia, CopyDataOptions copyDataOptions = CopyDataOptio
...
+                return 1;
+            }
+
+            var other = obj as Media;
+            if (other != null)
+            {
+                return Order.CompareTo(other.Order);
+            }
+            throw new ArgumentException("Object is not of type Media", nameof(obj));
+
```
Comment (PureKrome) [inline Code/OpenRealEstate.Core/Models/Media.cs:101]: Delete line.

File: Code/OpenRealEstate.Tests/Models/MediaTests.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+            //purposely put these in jumbled order
+            var mediaList = new List<Media>
+            {
+                media2,
+                media3,
+                media1
+            };
+
+            //Assert jumbed order
```
Comment (PureKrome) [inline Code/OpenRealEstate.Tests/Models/MediaTests.cs:29]: Don't need these 3x asserts.

File: Code/OpenRealEstate.Tests/Models/MediaTests.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+using Xunit;
+
+namespace OpenRealEstate.Tests.Models
+{
+    public class MediaTests
+    {
+        [Fact]
+        public void MediaShouldSortByValueOfOrderProperty()
+        {
+            var media1 = new Media { Order = 1 };
```
Comment (PureKrome) [inline Code/OpenRealEstate.Tests/Models/MediaTests.cs:17]: Add the text `// Arrange.` above this line.

e.g.

```
// Arrange.
var media1 = new M....
```

File: Code/OpenRealEstate.Tests/Models/MediaTests.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            var media2 = new Media { Order = 2 };
+            var media3 = new Media { Order = 3 };
+
+            //purposely put these in jumbled order
```
Comment (PureKrome) [inline Code/OpenRealEstate.Tests/Models/MediaTests.cs:21]: space between `//` and `P`.
`// Purpos....`

File: Code/OpenRealEstate.Tests/Models/MediaTests.cs (hunk 3 above)
Comment (PureKrome) [inline Code/OpenRealEstate.Tests/Models/MediaTests.cs:29]: Space and `.`

File: Code/OpenRealEstate.Tests/Models/MediaTests.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                media3,
+                media1
+            };
+
+            //Assert jumbed order
+            mediaList[0].Order.ShouldBe(2);
+            mediaList[1].Order.ShouldBe(3);
+            mediaList[2].Order.ShouldBe(1);
+
+            //Act
```
Comment (PureKrome) [inline Code/OpenRealEstate.Tests/Models/MediaTests.cs:34]: Space and .

File: Code/OpenRealEstate.Tests/Models/MediaTests.cs [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            mediaList.Sort();
+
+            //Assert
```
Comment (PureKrome) [inline Code/OpenRealEstate.Tests/Models/MediaTests.cs:37]: Space and .

File: Code/OpenRealEstate.Tests/Models/MediaTests.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace OpenRealEstate.Tests.Models
+{
+    public class MediaTests
+    {
+        [Fact]
+        public void MediaShouldSortByValueOfOrderProperty()
```
Comment (PureKrome) [inline Code/OpenRealEstate.Tests/Models/MediaTests.cs:15]: Rename: `GivenACollectionOfMedias_CompareTo_CorrectlySortsTheCollection`

**Review by PureKrome — CHANGES_REQUESTED** [review]

**Comment by PureKrome** [comment]
Also, change the test from a `[Fact]` to:

```
[Theory]
[MemberData(nameof(ValidData))]
public void GivenACollectionOfMedias_CompareTo_CorrectlySortsTheCollection(Media[] medias)
{
```

and make `ValidData` something like...

```
public static IEnumerable<object[]> ValidData => new[]
{
    new object[]
    {
        // Collection sorted.
    },
    new object[]
    {
        // Collection but unsorted.
    }
};
```


Also, need some tests for an empty collection (is that considered a valid test?)

also tests for comparing things that are not the same. This checks for the `ArgumentException`.

eg. Order vs a Cow or whatever.

**Comment by ctolkien** [comment]
>Also, need some tests for an empty collection (is that considered a valid test?)

How would we test that? That it doesn't throw?

**Comment by PureKrome** [comment]
> How would we test that? That it doesn't throw?

Because that's a valid scenario and whatever the correct solution is (under that scenario), needs to be 
1) considered
2) tested.

**Comment by PureKrome** [comment]
Manually ended up adding in these changes (or similar) to #42 .
